test/core/vm/actor/builtin/types/miner/monies_test_v0.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include "testutil/outcome.hpp"
+#include "vm/actor/builtin/types/miner/v0/monies.hpp"
+
+namespace fc::vm::actor::builtin::v0::miner {
+  class MoniesTestv0 : public testing::Test {
+   protected:
+    void SetUp() override {
+      nv = version::NetworkVersion::kVersion0;
+      epoch_target_reward = TokenAmount{BigInt(1) << 50};
+      sector_power = StoragePower{BigInt(1) << 36};
+      network_qa_power = StoragePower{BigInt(1) << 50};
+
+      reward_estimate = {.position = epoch_target_reward << 128,
+                         .velocity = BigInt(0) << 128};
+      power_estimate = std::make_shared<FilterEstimate>(FilterEstimate{
+          .position = network_qa_power << 128, .velocity = BigInt(0) << 128});
+
+      EXPECT_OUTCOME_TRUE(
+          maybe_penalty,
+          moniesv0.pledgePenaltyForUndeclaredFault(
+              reward_estimate, power_estimate, sector_power, nv));
+      undeclared_penalty_ = TokenAmount{maybe_penalty};
+      big_initial_pledge_factor = 20;
+    }
+
+    Monies moniesv0;
+
+    version::NetworkVersion nv;
+
+    TokenAmount epoch_target_reward;
+    StoragePower sector_power;
+    StoragePower network_qa_power;
+
+    FilterEstimate reward_estimate;
+    std::shared_ptr<FilterEstimate> power_estimate;
+
+    TokenAmount undeclared_penalty_;
+    BigInt big_initial_pledge_factor;
+  };
+
+  TEST_F(MoniesTestv0, TestPledgePenaltyForTermination) {
+    TokenAmount initial_pledge = TokenAmount{1 << 10};
+    TokenAmount day_reward = initial_pledge / big_initial_pledge_factor;
```

Comment:
bigdiv?

Reply:
what the difference to use common division or bigdiv in test where all values are predefined and do not lay in negative scope

Reply:
To be consistent. bigdiv() for BigInt should be used always